docs/content/rust/guides/roles/getting-started.md

```diff
@@ -0,0 +1,274 @@
+---
+title: Getting started
+weight: 2
+description: >
+    Get started with Oso Roles for Rust
+---
+
+[rust-polar-classes]: reference/polar/classes.html
+
+# Getting started
+
+When managing access to resources within an application, it can be useful to
+group permissions into **roles** and assign these roles to users. This is
+known as **Role-Based Access Control (RBAC).** The Oso library
+comes with built-in configuration for role-based access control.
+
+In this guide, we'll walk through the basics of starting to use the
+roles feature.
+
+## Setting up the Oso instance
+
+First, we'll cover some of the basics of integrating Oso into your
+application.
+
+The `oso::Oso` struct is the entrypoint to using Oso in our application.
+We usually will have a global instance that is created
+during application initialization and shared across requests.
+
+```rust
+use oso::Oso;
+
+fn init_oso() -> Oso {
+  let oso = Oso::new();
+```
+
+### Using Rust types with Oso
+
+Next, register any Rust types you'll be using in your policies.
+
+{{% callout "Rust type configuration" "blue" %}}
+
+Rust structs and enums will need
+[some extra configuration][rust-polar-classes] to work with Oso.
+
+{{% /callout %}}
+
+
+```rust
+  // register classes used by the policy
+  oso.register_class(User::get_polar_class());
+  oso.register_class(Org::get_polar_class());
+  oso.register_class(OrgRole::get_polar_class());
+```
+
+### Loading our policy
+
+Oso uses the [Polar language](/reference/polar/polar-syntax) to define authorization
+policies. An authorization policy specifies what requests are allowed and what data a
+user can access. The policy is stored in a Polar file, along with your code.
+
+Load the policy at application start using the `Oso::load_file()` function:
+
+```rust
+  // load the policy from a file
+  oso.load_file('app/policy/authorization.polar');
+```
+
+### Enable Oso Roles
+
+In order to enable the built-in roles features, we call the
+`Oso::enable_roles` function:
+
+```rust
+  oso.enable_roles();
+  oso
+}
```

Comment:
This is consistent with our other guides as it is now so this is probably fine, but what do you think about including the whole `init_oso` method in one snippet to make it easier to copy/paste?

I think it's particularly noticeable in rust because the snippets are each a "part" of the function and include unmatched curly braces.

Reply:
👍